arangod/Graph/Enumerators/WeightedTwoSidedEnumerator.cpp

```diff
@@ -824,14 +838,6 @@ typename WeightedTwoSidedEnumerator<QueueType, PathStoreType, ProviderType,
                                     PathValidator>::BallSearchLocation
 WeightedTwoSidedEnumerator<QueueType, PathStoreType, ProviderType,
                            PathValidator>::getBallToContinueSearch() {
-  if (!_leftInitialFetch) {
-    _leftInitialFetch = true;
-    return BallSearchLocation::LEFT;
-  } else if (!_rightInitialFetch) {
-    _rightInitialFetch = true;
-    return BallSearchLocation::RIGHT;
-  }
-
   if (!_left.isQueueEmpty() && !_right.isQueueEmpty()) {
     auto leftStep = _left.peekQueue();
     auto rightStep = _right.peekQueue();
```

Comment:
nit: `peekQueue()` returns a reference, so these lines will copy the `Step` objects.
I don't think that the copies are expensive here (looks like `Step` is a 24 byte object in most cases), but as `Step` is a template parameter, we may at some point have `Step`s which are more expensive to copy.
In this case it may be better to change `auto leftStep` and `auto rightStep` to `auto const& ...`.